home_work_container_logging/Logger.h

```diff
@@ -0,0 +1,116 @@
+//
+// Created by flyon21 on 09.11.24.
+//
+
+#ifndef C_PRO_LOGGER_H
+#define C_PRO_LOGGER_H
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <chrono>
+#include <map>
+#include <mutex>
+#include <iomanip>
+#include <vector>
+#include <mutex>
+
+
+class Logger {
+public:
+    enum LogLevel {
```

Comment:
Would be better to use class enum (or scoped enum).
enum class LogLevel {
INFO, WARNING, ERROR
}. Because in this case we will have such advanteges:
1) enum class does not implicitly convert to integers. This behaviour prevents accidental usage of enum values as integers or mixing different enum types unintentionally
2) enum values are scoped within the enum